### PR TITLE
Update Documentation for GCP Cloud SQL Client Authentication

### PR DESCRIPTION
**Pull request by jimbishopp**
Update docs for #9991.

### CODE REVIEW

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:70]:
```suggestion
  - "Cloud SQL Viewer" will allow the Database Service to automatically download
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:71]:
```suggestion
    your Cloud SQL instance's root CA certificate, but does not support client
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:73]:
```suggestion
  - "Cloud SQL Admin" will allow the Database Service to automatically download
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:74]:
```suggestion
    your Cloud SQL instance's root CA certificate and generate an
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:76]:
```suggestion
    "Allow only SSL connections."
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:98]:
```suggestion
  # is configured to "Allow only SSL connections."
```

File: docs/pages/database-access/guides/mysql-cloudsql.mdx
Comment (r0mant) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:38]: Should we say this one is "optional" then?

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (r0mant) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:99]: Same as above - mention this one is optional?

**Review by r0mant — COMMENTED** [review]
@jimbishopp @ptgott From the information organization perspective - do you think it'd make sense to also have a separate section in these guides covering specifically "enforce ssl connections" use-case? It would just explain in a couple of sentences that you need an additional GCP permission to generate client certs and the Teleport behavior in this case (that it will check itself whether instance forces ssl connections and attempt to generate client cert).

Reason is, I'm thinking if somebody asks about this use-case, we'll have a link we could give them. Right now it's sort of integrated in the guides which I think is fine but having a dedicated section might also be useful IMO.

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (jimbishopp) [reply docs/pages/database-access/guides/postgres-cloudsql.mdx:70]: This was copied from the previous release. There is another bullet above that has the same issue. I'll update everything.

File: docs/pages/database-access/guides/mysql-cloudsql.mdx
Comment (jimbishopp) [reply docs/pages/database-access/guides/mysql-cloudsql.mdx]: I had this originally. Don't remember why I removed it. Added it back.

**Comment by jimbishopp** [comment]
> Do you think it'd make sense to also have a separate section in these guides covering specifically "enforce ssl connections" use-case? 

@ptgott @r0mant I went into this thinking I was going to add a new section, but decided to change the existing sections because I couldn't decide where the new section would go. It also felt inconsistent to have another section describing the permissions needed. I also thought about creating a new box specifically for this but realized the appropriate place would be near the existing "SERVICE ACCOUNT PERMISSIONS" box. Adding a new box seemed redundant. I'm open to suggestion.

Keep in mind that if an existing user enables client authentication on the GCP instance without setting the appropriate permissions, they will receive an error messaging when trying to connect informing them of the permission they are missing.

**Comment by ptgott** [comment]
@r0mant @jimbishopp I think the `Admonition` box makes sense, since it provides context for assigning roles to the Database Service service account when we provide those instructions. If we want to make sure customers can view the box, what would you think of modifying the `Admonition` component to generate an ID that we could add to the URL? Currently the component has no `id`.

Just to check, would most users be fine with assigning the "Service Account Token Creator", "Cloud SQL Viewer", and "Cloud SQL Admin" roles without modification? The current use of the `Admonition` box implies that assigning more restrictive permissions to a new role is a special case.

**Review by r0mant — APPROVED** [review]

**Comment by jimbishopp** [comment]
@r0mant @ptgott I may have found a simple solution for having a link to send users. I added `H3` headers to break the "Create Service Account" section into steps. I like how the headers make each step easier to see and now we have a link specifically for "Grant Permissions" we could send users. WDYT?

> Just to check, would most users be fine with assigning the "Service Account Token Creator", "Cloud SQL Viewer", and "Cloud SQL Admin" roles without modification? The current use of the `Admonition` box implies that assigning more restrictive permissions to a new role is a special case.

I would think most users would want to implement the most restrictive permissions required for production. "Cloud SQL Admin" provides more than they need, but makes the guide less complex and easier for a new user to get through quickly when evaluating Teleport. Unfortunately, the new `cloudsql.sslCerts.createEphemeral` permission is only available for built-in GCP roles, of which "Cloud SQL Admin" is the most restrictive. Roman suggested we try to update these guides so they are still accurate if and when GCP allows the ephemeral permission to be assigned to custom roles in the future.

**Review by r0mant — APPROVED** [review]
@jimbishopp I think it's fine.

@ptgott Can you take another look as well so we can merge and get this backported/deployed?

File: docs/pages/database-access/guides/mysql-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:8]:
```suggestion
## Create a service account for the Teleport Database Service
```
Let's use sentence case for headings

File: docs/pages/database-access/guides/mysql-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:15]:
```suggestion
### Create a service account
```

File: docs/pages/database-access/guides/mysql-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:22]:
```suggestion
### Grant permissions
```

File: docs/pages/database-access/guides/mysql-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:46]:
```suggestion
### Create a key for the service account
```
To use sentence case and be a bit more descriptive

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:29]:
```suggestion
## Create a service account for the database
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:33]:
```suggestion
### Create a service account
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:42]:
```suggestion
### Grant permissions
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:52]:
```suggestion
### Configure authentication for your service account
```
To make this cohere a bit more with the sequence of steps we've outlined

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:67]:
```suggestion
## Create a service account for Teleport Database Service
```
Let's use sentence case here

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:73]:
```suggestion
### Create a service account
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:80]:
```suggestion
### Grant permissions
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:117]:
```suggestion
### Create a key for the service account
```
To use sentence case and be a bit more descriptive

**Review by ptgott — APPROVED** [review]
I've left minor suggestions. Since these are all suggestions, I have approved so you can merge right after you implement them.

**Pull request merged**
